boost-common/src/main/java/io/openliberty/boost/common/boosters/JDBCBoosterConfig.java

```diff
@@ -83,44 +82,24 @@ private void init(Properties boostConfigProperties, String configuredDatabaseDep
         // Set server properties
         this.serverProperties = new Properties();
 
+        // Initialize defaults and required properties for each datasource vendor
         if (this.dependency.startsWith(DERBY_DEPENDENCY)) {
-            String databaseName = (String) boostConfigProperties.getOrDefault(BoostProperties.DATASOURCE_DATABASE_NAME,
-                    DEFAULT_DERBY_DATABASE_NAME);
-            this.serverProperties.put(BoostProperties.DATASOURCE_DATABASE_NAME, databaseName);
-
+            this.serverProperties.put(BoostProperties.DATASOURCE_DATABASE_NAME, DEFAULT_DERBY_DATABASE_NAME);
+            this.serverProperties.put(BoostProperties.DATASOURCE_CREATE_DATABASE, "create");
         } else if (this.dependency.startsWith(DB2_DEPENDENCY)) {
-
-            // If serverName or portNumber are not provided, use DB2's default
-            // "localhost"
-            // and "50000"
-            String serverName = (String) boostConfigProperties.getOrDefault(BoostProperties.DATASOURCE_SERVER_NAME,
-                    LOCALHOST);
-            this.serverProperties.put(BoostProperties.DATASOURCE_SERVER_NAME, serverName);
-
-            String portNumber = (String) boostConfigProperties.getOrDefault(BoostProperties.DATASOURCE_PORT_NUMBER,
-                    "50000");
-            this.serverProperties.put(BoostProperties.DATASOURCE_PORT_NUMBER, portNumber);
-
-            // For databaseName, user, and password, there is no default that
-            // would serve
-            // any purpose. The variables
-            // will be configured in the server.xml, but no values will be set
-            // in
-            // bootstrap.properties. The values will
-            // need to be passed by the user at runtime.
-            String databaseName = (String) boostConfigProperties.get(BoostProperties.DATASOURCE_DATABASE_NAME);
-            if (databaseName != null) {
-                this.serverProperties.put(BoostProperties.DATASOURCE_DATABASE_NAME, databaseName);
-            }
-
-            String user = (String) boostConfigProperties.get(BoostProperties.DATASOURCE_USER);
-            if (user != null)
-                this.serverProperties.put(BoostProperties.DATASOURCE_USER, user);
-
-            String password = (String) boostConfigProperties.get(BoostProperties.DATASOURCE_PASSWORD);
-            if (password != null) {
-                this.serverProperties.put(BoostProperties.DATASOURCE_PASSWORD, password);
-            }
+        	this.serverProperties.put(BoostProperties.DATASOURCE_DATABASE_NAME, "");
```

Comment:
Sort of.  So these are all minimum required datasource properties for both embedded Derby and DB2. Both require a databaseName, and for embedded Derby, we can set this since we are creating the database for the user (by default).  For DB2, we aren't creating a DB, but expecting it to exist already, so there's no default value we can set that would be of any use. 
Conversely, DB2 requires a user and password, which for Derby is optional. And DB2 also requires a portNumber and serverName which embedded Derby I dont think uses at all... 
I will add some better comments for these.